pytransform3d/batch_rotations.py

```diff
@@ -38,6 +38,52 @@ def norm_vectors(V, out=None):
     return out
 
 
+def norm_axis_angles(a):
+    """Normalize axis-angle representation.
+
+    Parameters
+    ----------
+    a : array-like, shape (..., 4)
+        Axis of rotation and rotation angle: (x, y, z, angle)
+
+    Returns
+    -------
+    a : array, shape (..., 4)
+        Axis of rotation and rotation angle: (x, y, z, angle). The length
+        of the axis vector is 1 and the angle is in [0, pi). No rotation
+        is represented by [1, 0, 0, 0].
+    """
+    a = np.asarray(a)
+
+    # Handle the case of only one axis-angle instance
+    only_one = a.ndim == 1
+    a = np.atleast_2d(a)
+
+    angles = a[..., 3]
+    norm = np.linalg.norm(a[..., :3], axis=-1)
+
+    zero_mask = (angles == 0.0) | (norm == 0.0)
+    non_zero_mask = ~zero_mask
+
+    res = np.empty_like(a)
+    res[zero_mask, :] = np.array([1.0, 0.0, 0.0, 0.0])
+    res[non_zero_mask, :3] = (
+        a[non_zero_mask, :3] / norm[non_zero_mask, np.newaxis]
+    )
+
+    angle_normalized = norm_angle(angles)
+
+    negative_angle_mask = angle_normalized < 0.0
+    res[negative_angle_mask, :3] *= -1.0
+    angle_normalized[negative_angle_mask] *= -1.0
+
+    res[non_zero_mask, 3] = angle_normalized[non_zero_mask]
+
+    if only_one:
```

Comment:
Consider define the variable next to its usage - during code reading a reader/reviewer has to [search/scroll](https://softwareengineering.stackexchange.com/a/113264) up (ok maybe in this case this is not very critical) ...

```suggestion
    only_one = a.ndim == 1
    if only_one:
```

Reply:
We have to do it in the beginning, since the shape of a might change with `a = np.atleast_2d(a)`. The idea is to return only one 1D array, when we pass one 1D array to the function.